core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java

```diff
@@ -213,16 +213,19 @@ String formatStepText(String keyword, String stepText, Format textFormat, Format
         for (Argument argument : arguments) {
             // can be null if the argument is missing.
             if (argument.getOffset() != null) {
-                String text = stepText.substring(textStart, argument.getOffset());
-                result.append(textFormat.text(text));
+                int textEnd = argument.getOffset();
+                if (textStart <= textEnd ) { // textStart can be > textEnd with nested capture groups
+                    String text = stepText.substring(textStart, textEnd);
+                    result.append(textFormat.text(text));
+                }
             }
             // val can be null if the argument isn't there, for example @And("(it )?has something")
             if (argument.getVal() != null) {
                 result.append(argFormat.text(argument.getVal()));
                 textStart = argument.getOffset() + argument.getVal().length();
```

Comment:
@mpkorstanje Quick question to make sure I really understand: In your example, the second (nested) argument starts 17 positions before the end of the first (enclosing) argument. The offset would therefore be -17, right?